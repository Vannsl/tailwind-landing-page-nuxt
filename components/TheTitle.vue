<template>
  <div class="the-title" :class="{ light: isLight }">
    <h1>{{ label }}</h1>
    <div class="line-wrapper">
      <div class="line gradient"></div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'the-title',
  props: {
    label: String,
    isLight: {
      type: Boolean,
      default: false
    }
  }
}
</script>

<style scoped>
h1 {
  @apply w-full my-2 text-5xl font-bold leading-tight text-center text-gray-800;
}

.line-wrapper {
  @apply w-full mb-4;
}

.line {
  @apply h-1 mx-auto w-64 opacity-25 my-0 py-0 rounded-t;
}

.light h1 {
  @apply text-white;
}

.light .line {
  background: white !important;
}
</style>
