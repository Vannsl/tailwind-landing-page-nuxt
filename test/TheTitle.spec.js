import { shallowMount } from '@vue/test-utils'
import TheTitle from '@/components/TheTitle'

const factory = () => {
  return shallowMount(TheTitle)
}

describe('TheTitle', () => {
  it('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
