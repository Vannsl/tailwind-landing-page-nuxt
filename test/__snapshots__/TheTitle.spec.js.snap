// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheTitle renders properly 1`] = `"<div class=\\"the-title\\"><h1></h1> <div class=\\"line-wrapper\\"><div class=\\"line gradient\\"></div></div></div>"`;
